test/unit/template_importer_test.rb

```diff
@@ -0,0 +1,96 @@
+require 'test_plugin_helper'
+
+module ForemanTemplates
+  class TemplateImporterTest < ActiveSupport::TestCase
+    def get_template(name)
+      File.expand_path(File.join('..', '..', 'templates', name), __FILE__)
+    end
+
+    def importer(opts = {})
+      # This uses a checkout of the plugin as a source of templates Caveat:
+      # changes to /test/templates will need to be committed (locally) for
+      # tests to work
+
+      path = File.expand_path(File.join('..','..','..'),__FILE__)
+      ForemanTemplates::TemplateImporter.new({
+        :repo      => path,
+        :branch    => Git.open(path).current_branch,
+        :prefix    => 'FooBar ',
+        :dirname   => '/test/templates',
+        :verbose   => 'false',
+        :associate => 'new'
+      }.merge(opts))
+    end
+
+    setup do
+      @importer = importer
+    end
+
+    context 'get_default_branch' do
+      setup do
+        @repo = Struct.new(:branches).new([
+          Struct.new(:name).new('0.1-stable'),
+          Struct.new(:name).new('develop')
+        ])
+      end
+
+      test 'when on develop, returns develop' do
+        SETTINGS[:version].stubs(:tag).returns('develop')
+        assert_equal 'develop', @importer.get_default_branch(@repo)
+      end
+
+      test 'when branch exists, return it' do
+        SETTINGS[:version].stubs(:tag).returns('not_develop')
+        SETTINGS[:version].stubs(:short).returns('0.1')
+        assert_equal '0.1-stable', @importer.get_default_branch(@repo)
+      end
+
+      test 'when branch does not exist, use default branch' do
+        SETTINGS[:version].stubs(:tag).returns('not_develop')
+        SETTINGS[:version].stubs(:short).returns('0.2')
+        refute @importer.get_default_branch(@repo)
+      end
+    end
+
+    test 'metadata method extracts correct metadata' do
+      text = File.read(get_template('metadata1.erb'))
+      hash = @importer.parse_metadata(text)
+      assert_equal 'provision', hash['kind']
+      assert_equal 'Test Data', hash['name']
+      assert_equal 5, hash['oses'].size
+
+      text = File.read(get_template('metadata2.erb'))
+      hash = @importer.parse_metadata(text)
+      assert_equal 'ProvisioningTemplate', hash['model']
+    end
+
+    test 'import! loads templates into db' do
```

Comment:
Once you've got `purge!` set up in the importer class, we'll want to add a test similar to this one with 2 assertions. You'll want to create a couple of templates (something like `FactoryGirl.create(:provisioning_template, :name => 'blah')` should do), one with a prefix, one without. You can the call `.purge!` and assert that the correct one is gone and the other is still present.